test/puppetlabs/puppetdb/integration/soft_fail/soft_write_fail.clj

```diff
@@ -0,0 +1,26 @@
+(ns puppetlabs.puppetdb.integration.soft-fail.soft-write-fail
+  (:require [clojure.test :refer :all]
+            [puppetlabs.puppetdb.integration.fixtures :as int]
+            [puppetlabs.trapperkeeper.app :as tk-app]))
+
+(deftest ^:integration soft-write-fail
+  (with-open [pg (int/setup-postgres)
+              pdb (int/run-puppetdb pg {})
+              ps (int/run-puppet-server [pdb] {:terminus {:main {:soft_write_failure true}}})]
+    (tk-app/stop (-> pdb int/info-map :app))
+
+    (testing "Agent run should fail for manifest which collects resources"
+      (try
+        (int/run-puppet ps pdb "Notify <<| |>>")
+        (is false "The puppet run should have thrown an exception")
+        (catch clojure.lang.ExceptionInfo e
+          (is (re-find #"Could not retrieve resources from the PuppetDB"
+                       (-> e .getData :err))))))
+
+    (testing "Agent run should succeed for manifest which exports resources"
+      (try
+        (int/run-puppet ps pdb "@@notify { 'exported notify': }"
```

Comment:
This try/catch is in place because run-puppetdb always waits for commands to be processed. With puppetdb stopped, they will of course never be processed.